### ISSUE TITLE
Expose IP information of the hostname in the JSON output

### ISSUE THREAD

**Issue by p-l-** [projectdiscovery/nuclei#1220]
After #468, an `"ip"` field has been introduced to the JSON output for HTTP templates. The idea is to store which IP address has been used when several exist.

For non-HTTP templates (e.g., the `"network"` template `"CVE-2018-15473"`, it is not the case (see output example below).

In [IVRE](https://github.com/ivre/ivre) (and probably in several other tools), keeping the IP address used for the connection would be really helpful, either in case of an IP address change, or in case of several IP addresses existing for one hostname.

What do you think?

Here is a sample output for `"CVE-2018-15473"` template (provided by @0xtavian):
```JSON
{
  "template-id": "CVE-2018-15473",
  "info": {
    "name": "OpenSSH Username Enumeration",
    "author": [
      "r3dg33k",
      "daffainfo"
    ],
    "tags": [
      "network",
      "openssh",
      "cve",
      "cve2018"
    ],
    "description": "OpenSSH through 7.7 is prone to a user enumeration vulnerability due to not delaying bailout for an invalid authenticating user until after the packet containing the request has been fully parsed, related to auth2-gss.c, auth2-hostbased.c, and auth2-pubkey.c.",
    "reference": [
      "https://nvd.nist.gov/vuln/detail/cve-2018-15473"
    ],
    "severity": "medium",
    "classification": {
      "cve-id": [
        "cve-2018-15473"
      ],
      "cwe-id": [
        "cwe-362"
      ],
      "cvss-metrics": "CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:N/A:N",
      "cvss-score": 5.3
    }
  },
  "type": "network",
  "host": "scanme.nmap.org:22",
  "matched-at": "scanme.nmap.org:22",
  "timestamp": "2021-10-28T17:22:12.806348-05:00"
}
```

**Comment by Ice3man543** [comment]
This will be also handled under https://github.com/projectdiscovery/nuclei/issues/1218 where we'll define proper structures for interim and final output event. Thank you for reporting this @p-l-

**Comment by p-l-** [comment]
Actually the `"ip"` field has disappeared from Nuclei output, even in HTTP templates, somewhere between v2.5.1 (which has the field) and v2.5.2. I have not been able to spot more precisely this regression, because I cannot build several commits of Nuclei.

Thanks @itnsec for ivre/ivre#1203, I had totally missed that!

**Comment by p-l-** [comment]
@Ice3man543 @ehsandeep I think this should be labelled as "bug", as this is a regression: Nuclei v2.5.1 had the information in its JSON output.

**Comment by ehsandeep** [comment]
JSON output using **v2.5.1**:

```diff
{
  "templateID": "basic-example",
  "info": {
    "name": "Test HTTP Template",
    "author": [
      "pdteam"
    ],
    "tags": null,
    "reference": null,
    "severity": "info"
  },
  "type": "http",
  "host": "https://www.example.com",
  "matched": "https://www.example.com",
-  "ip": "93.184.216.34",
  "timestamp": "2021-11-22T05:01:41.457373+05:30"
}
```

JSON output using **v2.5.3-dev**:

```diff
{
  "template-id": "basic-example",
  "info": {
    "name": "Test HTTP Template",
    "author": [
      "pdteam"
    ],
    "tags": null,
    "reference": null,
    "severity": "info"
  },
  "type": "http",
  "host": "https://www.example.com",
  "matched-at": "https://www.example.com",
  "timestamp": "2021-11-22T05:01:57.949992+05:30",
+  "curl-command": "curl -X 'GET' -d '' -H 'Accept: */*' -H 'Accept-Language: en' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/49.0.2656.18 Safari/537.36' 'https://www.example.com'"
}
```

**Comment by Ice3man543** [comment]
This was traced to a missing option in the dialer which was causing it to not persist data regarding dialed IPs history. Should be fixed by #1273.
